src/core/useMutableSource.ts

```diff
@@ -68,7 +81,7 @@ export const useMutableSource = (
           ) {
             return prev
           }
-          if (prev[4] === nextSnapshot) {
+          if (Object.is(prev[4], nextSnapshot)) {
```

Comment:
Unlike valtio, jotai's snapshot can be anything, so use `Object.is` instead of `===`.